### PR TITLE
feat: admin user check middleware

### LINKED ISSUE #8389: [FEATURE] Middleware to project "admin" pages and api
### Description

Admin section is currently protected on each page route (inc. api) but they should be protected by NextJS middleware

- middleware protect `/admin/*`
- remove individual checks in admin files

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by bikrantjajware**
## Fixes Issue
Closes  #8389 

## Changes proposed

- Added middleware to check admin usernames on route "/admin" and "/api/admin"
- removed Admin username check from individual admin files

## Check List (Check all the applicable boxes) 

- [X ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [X ] All new and existing tests passed.
- [ X] This PR does not contain plagiarized content.
- [ X] The title of my pull request is a short description of the requested changes.

## Note to reviewers
NOTE:
- Using process.env in middleware file instead of serverEnv to avoid errors thrown while using Node APIs in edge runtime
- Did not remove getServerSession() from pages/admin/* files, because it would fail to read env variables if removed

### CODE REVIEW

File: middleware.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  const username = session.username;
+  if (!process.env.ADMIN_USERS.includes(username)) {
+    if (reqPathName.startsWith("/api")) {
+      return NextResponse.json({}, { status: 401 });
+    }
+    return NextResponse.redirect(new URL("/404", req.url));
+  }
+  req.cookies.set("usernames", username);
```
Comment (eddiejaoude) [inline middleware.js:25]: Why is this needed?

File: middleware.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  if (!session) {
+    if (reqPathName.startsWith("/api")) {
+      return NextResponse.json({}, { status: 401 });
+    }
+    return NextResponse.redirect(new URL("/auth/signin", req.url));
+  }
+
+  const username = session.username;
+  if (!process.env.ADMIN_USERS.includes(username)) {
+    if (reqPathName.startsWith("/api")) {
```
Comment (eddiejaoude) [inline middleware.js:20]: Should this be `/api/admin`?

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for the contribution 👍 

I have added some inline comments

File: middleware.js (hunk 1 above)
Comment (bikrantjajware) [reply middleware.js:25]: sorry this is not needed, will remove it. I was trying to set the username in cookies, so we can get it in the Component page without using getServerSession.

File: middleware.js (hunk 2 above)
Comment (bikrantjajware) [reply middleware.js:20]: ah we could add that but matcher config already does match for 
`  matcher: ["/admin/:path*", "/api/admin/:path*"]`
So i think it should be ok, but can update to /api/admin as well just to be sure

**Comment by bikrantjajware** [comment]
Hello @eddiejaoude,
just following up on this, do you have any comments :)

**Comment by eddiejaoude** [comment]
Sorry for my delay, I will have a look soon. It looks like the automated tests are passing which is great 👍

**Review by eddiejaoude — APPROVED** [review]
Thank you 👍

**Comment by bikrantjajware** [comment]
Thank you Eddie :)
